src/data/query.ts

```diff
@@ -0,0 +1,55 @@
+import { VectorOperationsApi } from '../pinecone-generated-ts-fetch';
+import type { QueryResponse } from '../pinecone-generated-ts-fetch';
+import { handleDataError } from './utils/errorHandling';
+import { buildConfigValidator } from '../validator';
+import { SparseValues } from './upsert';
+
+import { Static, Type } from '@sinclair/typebox';
+
+const nonemptyString = Type.String({ minLength: 1 });
+
+const shared = {
+  topK: Type.Number(),
+  includeValues: Type.Optional(Type.Boolean()),
+  includeMetadata: Type.Optional(Type.Boolean()),
+  filter: Type.Optional(Type.Object({})),
+};
+
+const QueryByVectorId = Type.Object({
+  ...shared,
+  id: nonemptyString,
+  vector: Type.Optional(Type.Never()),
+  sparseVector: Type.Optional(Type.Never()),
+});
+
+const QueryByVectorValues = Type.Object({
+  ...shared,
+  vector: Type.Array(Type.Number()),
+  sparseVector: Type.Optional(SparseValues),
+  id: Type.Optional(Type.Never()),
+});
+
+const Query = Type.Union([QueryByVectorId, QueryByVectorValues]);
+
+export type QueryByVectorId = Static<typeof QueryByVectorId>;
+export type QueryByVectorValues = Static<typeof QueryByVectorValues>;
+export type Query = Static<typeof Query>;
```

Comment:
**question / nit:** This came up while re-writing the delete API. Do we want to maintain the same naming convention for the typebox schema / options values as elsewhere? Specifically, sticking with the "-Schema" and "-Options" suffixes. Here we've got the `Query` schema object and also the `Query` exported static type.

Reply:
I like the -Options suffix. I haven't been 100% consistent on this but will try to do better in the future.